app/controlplane/internal/service/auth.go

```diff
@@ -289,6 +301,61 @@ func callbackHandler(svc *AuthService, w http.ResponseWriter, r *http.Request) (
 	return http.StatusTemporaryRedirect, nil
 }
 
+// autoOnboardOnOrganizations onboards the user to the organizations specified in the config
+// If the organization does not exist, it will be created with an inline CAS backend
+// If the user is not already a member of the organization, a membership will be created with the role specified in the config
+func autoOnboardOnOrganizations(ctx context.Context, svc *AuthService, u *biz.User) error {
+	for _, spec := range svc.onboardingConfig {
+		// Ensure the organization exists or create it if it doesn't
+		org, err := svc.orgUseCase.FindOrCreate(ctx, spec.GetName())
+		if err != nil {
+			return sl.LogAndMaskErr(err, svc.log)
+		}
+
+		// Determine the role to be assigned based on the config
+		role := determineRole(spec.GetRole())
+
+		// Ensures the user is a member of the organization with the specified role
+		if err := ensureMembership(ctx, svc, org.ID, u.ID, role); err != nil {
+			return sl.LogAndMaskErr(err, svc.log)
+		}
+	}
+	return nil
+}
+
+// determineRole maps the config role to the appropriate authz.Role
+func determineRole(configRole string) authz.Role {
```

Comment:
as mentioned before, we might not need this if we use the proto definitions that already exist + the mapping we already have.